scene/main/node.cpp

```diff
@@ -2907,7 +2919,7 @@ void Node::_bind_methods() {
 	BIND_VMETHOD(MethodInfo("_input", PropertyInfo(Variant::OBJECT, "event", PROPERTY_HINT_RESOURCE_TYPE, "InputEvent")));
 	BIND_VMETHOD(MethodInfo("_unhandled_input", PropertyInfo(Variant::OBJECT, "event", PROPERTY_HINT_RESOURCE_TYPE, "InputEvent")));
 	BIND_VMETHOD(MethodInfo("_unhandled_key_input", PropertyInfo(Variant::OBJECT, "event", PROPERTY_HINT_RESOURCE_TYPE, "InputEventKey")));
-	BIND_VMETHOD(MethodInfo(Variant::STRING, "_get_configuration_warning"));
+	BIND_VMETHOD(MethodInfo(Variant::ARRAY, "_get_configuration_warning"));
```

Comment:
Might be better to use a typed array `String[]` here (and `TypedArray<String>` in the C++).

Not sure how's the syntax to register this in `BIND_VMETHOD` as I'm not sure you can pass `PROPERTY_HINT_ARRAY_TYPE, "String"` to `MethodInfo` (normally it's used in `PropertyInfo`).

I found one instance of using `PropertyInfo` to qualify the return type of a `MethodInfo` here:
```
$ rg MethodInfo\\\(Propert
editor/editor_plugin.cpp
870:    ClassDB::add_virtual_method(get_class_static(), MethodInfo(PropertyInfo(Variant::OBJECT, "icon", PROPERTY_HINT_RESOURCE_TYPE, "Texture2D"), "get_plugin_icon"));
```

Not sure what `icon` is used for here, might just be a placeholder as you can't have an unnamed property.

Reply:
Note that it's perfectly possible to instantiate `PropertyInfo`/`MethodInfo` locally and set needed fields/members manually, since those are public (without using constructor syntax). See example in GDScript functions:

https://github.com/godotengine/godot/blob/a4f70d7e6cb2b83f152717a69c9c042dd9424fc9/modules/gdscript/gdscript_functions.cpp#L1867-L1872